github/spec/features/validate_repos_spec.rb

```diff
@@ -0,0 +1,88 @@
+require_relative '../spec_helper'
+require_relative '../../lib/configure_repos'
+require_relative '../../lib/fetch_repos'
+require_relative '../../lib/validate_repos'
+
+require 'base64'
+require 'yaml'
+require "octokit"
+require "json"
+require "open-uri"
+
+RSpec.describe ValidateRepos do
+
+  it "should ignore any repos that exist in repos.json AND are tagged govuk in GitHub" do
+    repos = [{
+      "app_name": "this-is-a-govuk-repo",
+    }]
+
+    repo_mock = double("Repo", topics: ["govuk"], archived: false)
+    allow(repo_mock).to receive(:[]).with("name").and_return("this-is-a-govuk-repo")
+    allow(repo_mock).to receive(:full_name).and_return("this-is-a-govuk-repo")
+    allow(repo_mock).to receive(:[]).with(:full_name).and_return("this-is-a-govuk-repo")
```

Comment:
Are both of these lines necessary?

Reply:
Unfortunately in its current state, yes. (Unless they can be somehow 'combed' into one), removing either gives me different errors.
Absence of line 21:  #<Double "Repo"> received unexpected message :full_name with (no args)
Absence of line 22: #<Double "Repo"> received :[] with unexpected arguments
         expected: ("name")
              got: (:full_name)

If you change the original line 20 to use (:full_name) instead of ("name") then it returns this same error, just the other way around. (expects :full_name, gets "name")